client/internal/connect.go

```diff
@@ -330,6 +331,12 @@ func createEngineConfig(key wgtypes.Key, config *Config, peerConfig *mgmProto.Pe
 		engineConf.PreSharedKey = &preSharedKey
 	}
 
+	port, err := freePort(config.WgPort)
+	if err != nil {
+		return nil, err
+	}
+	engineConf.WgPort = port
```

Comment:
Hello @mattkasun thanks for the contribution.

Can you add a log with info level to inform the user that another port will be used?